package PBData.Gxx;

message AppInfoRegion
{
	optional uint32 app_version = 2;
	optional uint32 client_type = 3;
	optional string region = 4;
	optional string download_path = 5;
	optional uint32 create_time = 6;
	optional uint32 update_time = 7;
}

message AppInfo
{
	optional uint32 app_version = 2;
	optional uint32 client_type = 3;
	optional string name = 4;
	optional string download_url = 5;
	optional bytes signature = 6;
	optional uint32 size = 7;
	optional string memo = 8;
	optional uint32 create_time = 9;
	optional uint32 update_time = 10;
}

message DataFile
{
	optional uint32 file_id = 2;
	optional uint32 file_version = 3;
	optional uint32 client_type = 4;
	optional string name = 5;
	optional string download_url = 6;
	optional string local_path = 7;
	optional bytes signature = 8;
	optional uint32 size = 9;
	optional uint32 min_app_version = 10;
	optional uint32 max_app_version = 11;
	optional string memo = 12;
	optional uint32 create_time = 13;
	optional uint32 update_time = 14;
}

message Config
{
	optional uint32 key = 2;
	optional string value = 3;
	optional uint32 create_time = 4;
}

message UserToken
{
	optional uint64 token_id = 1;
	optional uint64 uid = 2;
	optional string token = 3;
	optional uint32 create_time = 4;
	optional uint32 expiry_time = 5;
	optional uint32 client_type = 6;
	optional bytes device_id = 7;
}

message UserInfo
{
	optional uint64 uid = 1;
	optional string signature = 2;
	optional uint32 update_time = 3;
}

message DataVersion
{
	optional uint32 version_type = 1;
	optional uint64 version_id = 2;
	optional uint32 version = 3;
	optional uint32 create_time = 4;
}

message Tag
{
	optional uint64 tag_id = 1;
	optional uint64 uid = 2;
	optional string name = 3;
	optional uint32 create_time = 4;
	optional uint32 update_time = 5;
}

message Buddy
{
	optional uint64 uid = 2;
	optional uint64 buddy_uid = 3;
	optional string rename = 4;
	optional uint32 relation = 5;
	optional uint32 create_time = 6;
	optional uint32 update_time = 7;
}

message TagBuddy
{
	optional uint64 uid = 2;
	optional uint64 buddy_uid = 3;
	optional uint64 tag_id = 4;
	optional uint32 create_time = 5;
}

message TagBuddyImport
{
	optional uint64 uid = 2;
	optional uint64 buddy_uid = 3;
	optional uint64 game_id = 4;
	optional uint32 create_time = 5;
}

message BuddyPurge
{
	optional uint64 uid = 1;
	optional uint64 buddy_uid = 2;
	optional uint32 create_time = 3;
}

message Discussion
{
	optional uint64 discussion_id = 1;
	optional string name = 2;
	optional uint64 owner_uid = 3;
	optional uint32 max_member_count = 4;
	optional uint32 flag = 5;
	optional uint32 create_time = 6;
	optional uint32 update_time = 7;
}

message DiscussionMember
{
	optional uint64 discussion_id = 2;
	optional uint64 uid = 3;
	optional uint32 create_time = 4;
	optional uint32 update_time = 5;
}

message Group
{
	optional uint64 group_id = 1;
	optional uint32 group_category_id = 2;
	optional uint32 group_sub_category_id = 3;
	optional string name = 4;
	optional string description = 5;
	optional string announcement = 6;
	optional uint32 max_member_count = 7;
	optional bytes avatar_id = 8;
	optional uint32 points = 9;
	optional uint64 flag = 10;
	optional uint32 create_time = 11;
	optional uint32 update_time = 12;
}

message GroupMemberTitle
{
	optional uint64 group_id = 1;
	optional uint32 title_id = 2;
	optional string title = 3;
}

message GroupMember
{
	optional uint64 group_id = 1;
	optional uint64 uid = 2;
	optional uint32 role = 3;
	optional uint32 title_id = 4;
	optional string display_name = 5;
	optional uint32 create_time = 6;
	optional uint32 update_time = 7;
}

message Message
{
	optional uint32 session_type = 1;	// Constant.MessageSessionType
	optional uint64 session_id_lower = 2;
	optional uint64 session_id_upper = 3;
	optional uint64 message_id = 4;
	optional uint64 from_uid = 5;
	optional uint32 message_type = 6;
	optional bytes message = 7;
	optional bytes data = 8;
	optional uint32 flag = 9;	// Constant.MSMessageFlag
	optional bytes ip = 10;
	optional uint32 client_type = 11;
	optional uint32 create_time = 12;
	optional uint32 update_time = 13;
	optional uint64 unique_id = 14;
}

message MessageSession
{
	optional uint64 uid = 1;
	optional uint32 session_type = 2;  // Constant.MessageSessionType
	optional uint64 session_id_lower = 3;
	optional uint64 session_id_upper = 4;
	optional uint32 flag = 5;  // Constant.MessageSessionFlag
	optional uint64 start_message_id = 6;
	optional uint64 read_message_id = 7;
	optional uint32 create_time = 8;
	optional uint32 update_time = 9;
}

message MobileInfo
{
	optional uint64 uid = 1;
	optional string device_id = 2;
	optional uint32 client_type = 3;
	optional uint32 notify_token_type = 4;
	optional bytes notify_token = 5;
	optional string language = 6;
	optional uint32 status = 7;
	optional uint32 create_time = 8;
	optional uint32 update_time = 9;
}

message Request
{
	optional uint64 request_id = 1;
	optional uint32 request_type = 2;
	optional uint64 requester_uid = 3;
	optional uint64 target_id = 4;
	optional uint64 verifier_uid = 5;
	optional uint64 related_id = 6;
	optional uint32 status = 7;
	optional bytes request_message = 8;
	optional bytes reply_message = 9;
	optional bytes data = 10;
	optional uint64 related_request_id = 11;
	optional uint32 create_time = 12;
	optional uint32 update_time = 13;
}

message FileRequest
{
	optional uint64 request_id = 1;
	optional uint64 sender_uid = 2;
	optional uint32 target_type = 3;
	optional uint64 target_id = 4;
	optional bytes file_id = 5;
	optional string file_name = 6;
	optional uint64 file_size = 7;
	optional uint32 flag = 8;
	optional uint32 status = 9;
	optional uint32 create_time = 10;
	optional uint32 update_time = 11;
}

message UserSetting
{
	optional uint64 uid = 2;
	optional uint32 type = 3;
	optional uint32 value = 4;
	optional uint32 create_time = 5;
	optional uint32 update_time = 6;
}

message UserData
{
	optional uint64 uid = 2;
	optional uint32 key = 3;
	optional bytes value = 4;
	optional uint32 create_time = 5;
	optional uint32 update_time = 6;
}

message CastServer
{
	optional uint32 id = 1;
	optional string server_ip = 2;
	optional uint32 connection_port = 3;
	optional uint32 data_port = 4;
	optional string region = 5;
	optional uint32 create_time = 6;
	optional uint32 update_time = 7;
}

message TalkRoom
{
	optional uint64 room_id = 2;
	optional uint32 cast_server_id = 3;
	optional uint32 flag = 4;
	optional uint32 create_time = 5;
	optional uint32 update_time = 6;
}

message TalkRoomMember
{
	optional uint64 room_id = 2;
	optional uint64 uid = 3;
	optional uint32 role = 4;
	optional uint32 status = 5;
	optional uint32 create_time = 6;
	optional uint32 update_time = 7;
}

message Follower
{
	optional uint64 uid = 2;
	optional uint64 follower_uid = 3;
	optional uint32 app_id = 4;
	optional uint32 flag = 5;
	optional uint32 create_time = 6;
	optional uint32 update_time = 7;
}

message Followee
{
	optional uint64 uid = 2;
	optional uint64 followee_uid = 3;
	optional uint32 app_id = 4;
	optional uint32 flag = 5;
	optional uint32 create_time = 6;
	optional uint32 update_time = 7;
}
